tree/30_generic_methods/service_ensure_running_path.cf

```diff
@@ -36,22 +36,30 @@ bundle agent service_ensure_running_path(service_name, service_path)
 
     "class_prefix"                 string => "service_ensure_running_${canonified_service_name}";
 
+  classes:
+    # This method can be called as a substitution if service_ensure_running is called directly, with service_path=service_name
+    # In this case, we don't need to call service_check_running_ps but can use better OS-specific methods if available
+    "real_path"                       not => strcmp("${service_name}", "${service_path}");
```

Comment:
This is rather ugly. The problem is that ```service_ensure_running(x)``` calls ```service_ensure_running_path(x, x)```, so there is no other implementation than this one. That was not a problem until now, because with only one implementation, there were no choices to be made. But now, there is a "real" ```service_check_running``` method that uses src, not ps. So we need to make a choice... This is kinda like backpedalling, though.